src/PortfolioPlanning/Redux/Sagas/DefaultDateUtil.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    PortfolioPlanningFullContentQueryResult
+} from "../../Models/PortfolioPlanningQueryModels";
+import { JsonPatchDocument } from "VSS/WebApi/Contracts";
+import { IEpic } from "../../Contracts";
+import { WorkItemTrackingHttpClient } from "TFS/WorkItemTracking/RestClient";
+import * as VSS_Service from "VSS/Service";
+import { effects, SagaIterator } from "redux-saga";
+import { getEpicById } from "../Selectors/EpicTimelineSelectors";
+
+export function* SetDefaultDatesForEpics(queryResult: PortfolioPlanningFullContentQueryResult) {
+
+    let epicsWithoutDates: number[] = [];
+    let now, oneMonthFromNow;
+
+    queryResult.items.items.map(item => {
+        if (!item.StartDate || !item.TargetDate) {
+            now = new Date();
+            now.setHours(0, 0, 0, 0);
+            oneMonthFromNow = new Date();
+            oneMonthFromNow.setHours(0, 0, 0, 0);
+            oneMonthFromNow.setDate(now.getDate() + 30);
+            epicsWithoutDates.push(item.WorkItemId);
+            item.StartDate = now;
+            item.TargetDate = oneMonthFromNow;
+        }
+    });
+
+    // TODO: Add error handling.
+    yield effects.all(epicsWithoutDates.map(epic => effects.call(saveDatesToServer, epic, now, oneMonthFromNow)));
+}
+
+/*
+This method is called from two places:
+1. setting dates for a selected epic from UI
+2. set default dates for newly added epic.
+In second case, epic is not saved into state yet.
+*/
+function* saveDatesToServer(epicId: number, defaultStartDate?: Date, defaultEndDate?: Date): SagaIterator {
```

Comment:
Looks like this is still defined in EpicTimelineSaga. I think moving it here is the right call. Just delete it from the other file.

Reply:
Oh Yes, I intended to delete it.